index.js

```diff
@@ -30,26 +42,23 @@ class BindingComponent extends React.Component {
         val = this.props.bindStateType(val);
       }
     }
-    const names = this.props.bindStateName.split('.');
-    if (names.length === 1) {
-      this.props.bindStateCtx.setState({
-        [names[0]]: val
-      });
-    } else {
-      const newState = this.props.bindStateCtx.state[names[0]];
-      const subnames = names.slice(1);
-      subnames.reduce((curr, name, index) => {
-        if (index === subnames.length - 1) {
-          curr[name] = val;
-        } else {
-          curr = curr[name];
-        }
-        return curr;
-      }, newState);
-      this.props.bindStateCtx.setState({
-        [names[0]]: newState
-      });
-    }
+
+    this.props.bindStateCtx.replaceState((prevState) => {
```

Comment:
And since we are doing merging manually, there's no difference calling _Component.setState_ or _Component.replaceState_.

Note: After reading [documentation], I found _replaceState_ is **deprecate** so _setState_ would be a better choice.